frontend/js/src/playlists/PlaylistMenu.tsx

```diff
@@ -86,35 +97,32 @@ function PlaylistMenu({
       handleError(error);
     }
   };
-  const exportAsJSPF = async (
-    playlistId: string,
-    playlistTitle: string,
-    auth_token: string
-  ) => {
-    const result = await APIService.getPlaylist(playlistId, auth_token);
+
+  const exportAsJSPF = async () => {
+    const result = await APIService.getPlaylist(playlistID, auth_token);
     saveAs(await result.blob(), `${playlistTitle}.jspf`);
   };
 
-  const exportAsXSPF = async (
-    playlistId: string,
-    playlistTitle: string,
-    auth_token: string
-  ) => {
+  const exportAsXSPF = async () => {
+    if (!auth_token) {
+      alertMustBeLoggedIn();
+      return;
+    }
     const result = await APIService.exportPlaylistToXSPF(
       auth_token,
-      playlistId
+      playlistID
     );
     saveAs(result, `${playlistTitle}.xspf`);
   };
 
-  const exportToSpotify = async (
-    playlistId: string,
-    playlistTitle: string,
-    auth_token: string
-  ) => {
+  const exportToSpotify = async () => {
+    if (!auth_token) {
+      alertMustBeLoggedIn();
+      return;
+    }
     const result = await APIService.exportPlaylistToSpotify(
       auth_token,
-      playlistId
+      playlistID
```

Comment:
We have the same issue here that I mentioned earlier: this only works if there is an existing playlist on the server.
In the case of LBRadio playlists, they are ephemeral and `playlistID` will be undefined.
Instead, I guess we would first need to create a new playlist [as we do here](https://github.com/metabrainz/listenbrainz-server/pull/2588/files#diff-9789b1b5a9afea5adb31b3f150de1651f3bead3f988a6158a4e4ff4324a81978R71-R73) and use the new playlist ID to export it.